### DISCUSSION TITLE
[Feature] Passkeys

### DISCUSSION THREAD

**Discussion by uhthomas** [immich-app/immich#3338] [Feature Request]
### The feature

Passkeys are a method of authentication which does not require passwords. It's simpler to use than oauth, more secure than passwords and easier to manage. It's gaining wide adoption and support across the industry.

https://www.passkeys.io/
https://blog.google/technology/safety-security/the-beginning-of-the-end-of-the-password/
https://developer.apple.com/passkeys/
https://fidoalliance.org/passkeys/
https://developers.google.com/identity/passkeys

Maybe useful for how to integrate passkeys: https://www.passkeys.com/guides

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by j5k** [comment]
Also support (multiple) FIDO2/webauthn security token like YubiKeys

**Comment by zhiyan114** [comment]
Also looking for this!

If possible, also implement the standard 2-factor auth (U2F, TOTP) for users with limited keypass slots on hardware-based key (i.e. yubikey v5.4.3 or older only has 25 slots)

**Comment by eiqnepm** [comment]
Passkey support would be awesome! It would bypass the need for complicated third party 2FA solutions and I could use my security key as a hardware passkey—making authentication something I have and something I know.

**Comment by mmomjian** [comment]
we support OIDC for more advanced auth

**Reply by uhthomas** [reply]
I don't think OIDC and Passkeys are interchangeable. Does Immich actually support passkeys now?

**Reply by mmomjian** [reply]
You would use the passkey on the frontend auth provider. Immich does not have any fancy auth builtin and we plan to keep it that way.

**Reply by uhthomas** [reply]
That's a shame? I wouldn't categorise passkeys as "fancy". Two factor authentication is important and becoming ever prevalent. Although Immich is primarily targeted at self hosted, private instances, it would help anyone who wanted to expose Immich publicly and is a prerequisite for anyone looking to use Immich as a hosted cloud option.

Deferring security to OIDC when Immich has its own concept of authentication doesn't really make sense to me. OIDC is not a replacement for 2FA or passkeys.

Would you mind reopening this? I think it's important and the reactions clearly show it's a desirable feature.

**Reply by mmomjian** [reply]
We've discussed this at length, #8175 is one example, and we have decided we are not expanding the login scope of Immich. This lets us forus on building Immich and delegate auth to existing solutions.

**Reply by uhthomas** [reply]
Okay, thanks for the explanation and linking to existing discussions.

It's a shame and it's not something I agree with, but I respect your decision.

**Reply by mmomjian** [reply]
FWIW I would personally like TOTP support (the other discussion) but I understand the current limitations of the project scope. I setup fail2ban on the Immich logs for attempts and that lets me rest easy.

**Comment by LuckyCoders** [comment]
I'd still really like to see native options for:

1. **TOTP** — confirmation via an authenticator app code at login
2. **Passkeys / WebAuthn** — browser / hardware key (YubiKey, etc.) login

I understand the project preference is to keep Immich auth simple and push advanced MFA to OAuth/OIDC (Authentik, Authelia, …). That's a workable path, but for a small private instance the overhead of a full IdP is heavy compared to optional built-in TOTP or passkeys.

Just adding another +1 here for considering this again someday — even as optional, non-default features. Thanks for all the work on Immich.
